src/Flowpipes/reachsets.jl

```diff
@@ -585,10 +585,14 @@ function overapproximate(R::TaylorModelReachSet{N}, ::Type{<:Hyperrectangle}) wh
     # pick the time domain of the given TM (same in all dimensions)
     t0 = tstart(R)
     Δt = tspan(R)
+    # tdom defined below is the same as Δt - t0, but the domain inclusion check
+    # in TM.evauate may fail, so we unpack the domain again here; also note that
+    # by construction the TMs in time are centered at zero
+    tdom = TM.domain(first(set(R)))
 
     # evaluate the Taylor model in time
     # X_Δt is a vector of TaylorN (spatial variables) whose coefficients are intervals
-    X_Δt = TM.evaluate(set(R), Δt - t0)
+    X_Δt = TM.evaluate(set(R), tdom)
```

Comment:
Two questions:
- Is `t0` the expansion point, or is it `inf(tdom)`?
- Is `Δt` different from `tdom`?

One remark: I think the correct evaluation should read `TM.evaluate(set(R), tdom-t0)`, with `t0` the point where the Taylor expansion is constructed.

Reply:
The `R` here is `xTM1v` in the `validated_integ` algorithm, defined in this line: https://github.com/JuliaReach/ReachabilityAnalysis.jl/blob/824789cbd610bc88f0eb1ff831dd2da72d1f4f72/src/Algorithms/TMJets/validated_integ.jl#L530

For `xTM1v`, the expansions in time are centered at zero (`zI`). That's why `TM.evaluate(set(R), tdom-t0)` should match `TM.evaluate(set(R), tdom-tdom.x0)`. Here, `t0` corresponds to the *start time* of this reach-set, which is the final time of the previously computed reach-set.

Reply:
To be clear,

> Is t0 the expansion point, or is it inf(tdom)?

The expansion point of this taylor model is `inf(tdom)`, which is zero by construction in `validated_integ`. OTOH, `t0` is what we call the `tstart(R)`; think of a flowpipe as a union of reach-sets, and each reach-sets carries it's time span that you can access with `tspan(R)`; there is `tstart(R)` and `tend(R)` which are just the `inf` and `sup` of the interval `tspan(R)`. 

> Is Δt different from tdom?

Yes. `tdom` is the domain of the taylor model which by construction are always of the form `[0, t]`. However, `Δt = tspan(R)`.

---

Storing time like that allows to see flowpipes as "one thing", instead of the reach-sets in isolation; in other words, you can take a solution and ask for `sol(1 .. 2)`  which will return all those reach-sets whose time intervals intersect `1 .. 2`; and also do `sol(30.5)` and this will return the reach-set corresponding to time `30.5`.

Reply:
Thanks for the explanation; that clarifies a lot. 

One disadvantage of storing [t0, t1], [t1, t2], [t2, t3], ... is that you will loose precision if the integration span is large: simply think that `eps(tn)` grows with  a growing `tn`, so you loose some significant figures. That's the reason that `validated_integ` returns the vector `[t0, t1, t2, ...]`, and the `TaylorModel1`s spanning the interval [0, δtn].

Reply:
Interesting observation! As i did in this PR, one can access this "true" domain through `TM.domain(first(set(R)))`; i pick the first one because it's the same for all components.  Perhaps there should be a getter function for this.

Reply:
I am not sure how general this is, I mean, to have a vector of TM1s which share the same domain. I also think we should provide a getter function for `tm.x0`...

Reply:
:+1:  see PR https://github.com/JuliaReach/ReachabilityAnalysis.jl/pull/190.

> I also think we should provide a getter function for tm.x0...

i've opened https://github.com/JuliaIntervals/TaylorModels.jl/issues/75